Add ToolBox extension

kkange opened the pull request:
## Description

This is a collection extension of various string conversion scripts.
You can use the strings stored in the clipboard to execute the function or you can directly enter it and execute it.

Scripts are easy to add and we will continue to add more.

## Type of change

- New extension

## Screencast

[script list](https://imgur.com/XidJIIQ)
[form input](https://imgur.com/Vzqy0J4)
[direct input](https://imgur.com/pUfNOaY)


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

es183923 commented on extensions/toolbox/src/script/util.ts:
```diff
@@ -0,0 +1,6 @@
+import { execa } from "execa";
+
+export async function readClipboard() {
+  const { stdout } = await execa("pbpaste", ["unicorns"]);
```
why is there a `["unicorns"]` here?

kkange replied:
Thanks to your advice, I modified it.

es183923 commented:
I just want to say that some of the functions are already available in other extensions, such as the Came Case function, in the [change case](https://www.raycast.com/es183923/change-case) extension, and the random number function, in the [random data generator](https://www.raycast.com/loris/random) extension.

kkange commented:
> I just want to say that some of the functions are already available in other extensions, such as the Camel Case function, in the [change case](https://www.raycast.com/es183923/change-case) extension, and the random number function, in the [random data generator](https://www.raycast.com/loris/random) extension.


The toolbox extension program is a simple text conversion tool with various functions but few choices to set.

On the other hand, the extended programs you told me have a variety of options and various data, so I think they are different from the toolbox expansion.

es183923 commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,208 @@
+          )}
+          {(info.type === "all" || info.type === "clipboard") && (
+            <ActionPanel.Item
+              shortcut={{ modifiers: ["cmd"], key: "s" }}
```
I think cmd+shift-enter would be better here.

kkange replied:
Oh, that's right. If you change it like that, it'll be consistent.

es183923 commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,208 @@
+          {(info.type === "all" || info.type === "clipboard") && (
+            <ActionPanel.Item
+              shortcut={{ modifiers: ["cmd"], key: "s" }}
+              title={"Run Script - Window Keep"}
```
A better title would be `Run Script - Keep Window`

kkange replied:
Thank you. I fixed it.

es183923 commented on extensions/toolbox/src/script/date/timestamp.ts:
```diff
@@ -0,0 +1,26 @@
+export const toTimestamp: Script = {
+  info: {
+    title: "Date to Timestamp",
+    desc: "Converts dates to Unix timestamp",
```
`"Convert dates to Unix timestamps"` would be better here.

kkange replied:
Thank you. I fixed it.

es183923 commented on extensions/toolbox/src/script/date/utc.ts:
```diff
@@ -0,0 +1,26 @@
+export const toUtc: Script = {
+  info: {
+    title: "Date to UTC",
+    desc: "Converts dates and timestamps to UTC dates",
```
`Convert dates and timestamps to UTC dates` would be better here.

kkange replied:
Thank you. I fixed it.

es183923 commented on extensions/toolbox/src/script/string/camelCase.ts:
```diff
@@ -0,0 +1,9 @@
+import * as _ from "lodash";
+
+export const camelCase: Script = {
+  info: { title: "Camel Case", desc: "Converts Your Text To Camel Case", type: "all", example: "ray cast" },
```
`Converts Your Text To Camel Case` should be `Convert your text into Camel Case`.

kkange replied:
Thank you. I fixed it.

es183923 commented:
Instead of saying `Decodes` or `Converts`, it should just be `Decode` or `Convert`

kkange commented:
> Instead of saying `Decodes` or `Converts`, it should just be `Decode` or `Convert`

Thank you for helping me. You're a good person.

pernielsentikaer reviewed (changes requested):
Hi @kkange 👋 

I tested your extension and have a few comments:

- If having nothing or wrong format in clipboard while visiting “Date to Timestamp” and “Date to UTC” it flashes loading toast → error toast and change view.

- The UI is a bit confusing when nothing is explaining to me what to do, I didn't catch what I should do at first — the empty item could hold some help (I can also see that every view have a form view, I would stick with that one for the most of them instead)

- The loading toast when entering a script feels redundant to me, since it's not loading anything that should cause delay and only flashes quickly.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

kkange commented:
Thank you for the review.

I think I revised what you said.

pernielsentikaer commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,324 @@
+        <ActionPanel>
+          {(info.type === "all" || info.type === "clipboard") && (
+            <ActionPanel.Item
+              title={"Run Script Clipboard"}
```
```suggestion
              title={"Run Script to Clipboard"}
```

pernielsentikaer commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,324 @@
+          )}
+          {(info.type === "all" || info.type === "noclipboard" || info.type === "direct") && (
+            <ActionPanel.Item
+              title={"Run Script Direct"}
```
```suggestion
              title={"Run Script"}
```

pernielsentikaer commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,324 @@
+            icon={Icon.Clipboard}
+            onAction={async () => {
+              await copyTextToClipboard(content.result);
+              await showHUD("Copy Result");
```
```suggestion
              await showHUD("✅ Result Copied to Clipboard");
```

pernielsentikaer commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,324 @@
+            icon={Icon.Clipboard}
+            onAction={async () => {
+              await copyTextToClipboard(content.query);
+              await showHUD("Copy Query");
```
```suggestion
              await showHUD("✅ Query Copied to Clipboard");
```

pernielsentikaer commented on extensions/toolbox/src/index.tsx:
```diff
@@ -0,0 +1,324 @@
+        isClipboardScriptRunning = false;
+        toast.hide();
+        if (scriptResult.isSuccess) {
+          await showHUD("Copy Result");
```
```suggestion
          await showHUD("✅ Result Copied to Clipboard");
```

pernielsentikaer reviewed (changes requested):
Hi @kkange 👋 

Firstly, sorry for the longer response time, I had to try it out for a longer period than expected.
 
Thanks for the update, we're getting closer to release, I have some text corrections and comments:

- Adding extra delay makes the extension “slow” — since everything is happening locally there is no really need for loading spinners, I suggest removing extra delay and loading spinners.

Looking forward for the update 🔥 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

kkange commented:
I accepted your proposal and changed it.

And after using it for a few days,
It was convenient to change the basic action to the input screen. So, I changed it.

Action order.
1. Run Script
2. Run Script From
3. Run Script to Clipboard

pernielsentikaer reviewed (approved):
Hi @kkange 👋 

I'm going to accept it now, I think its good as it is right now 💯 

Thanks for you contribution 👍
